test/logging-engine.js

```diff
@@ -0,0 +1,166 @@
+/* eslint-env mocha */
+/* eslint-disable no-unused-expressions */
+const chai = require('chai');
+const sinon = require('sinon');
+const sinonChai = require('sinon-chai');
+chai.should();
+chai.use(sinonChai);
+
+const { start, dispatch, stop, spawnStateless } = require('../lib');
+const {
+  LogLevel,
+  LogEvent,
+  logNothing,
+  configureLogging
+} = require('../lib/monitoring');
+const {
+  logLevelToString,
+  LoggingFacade,
+  LoggingFacadeImpl
+} = require('../lib/monitoring/logging-engine');
+
+describe('LogLevel', () => {
+  it('log levels should be sorted', () => {
+    LogLevel.OFF.should.be.lessThan(LogLevel.TRACE);
+    LogLevel.TRACE.should.be.lessThan(LogLevel.DEBUG);
+    LogLevel.DEBUG.should.be.lessThan(LogLevel.INFO);
+    LogLevel.INFO.should.be.lessThan(LogLevel.WARN);
+    LogLevel.WARN.should.be.lessThan(LogLevel.ERROR);
+    LogLevel.ERROR.should.be.lessThan(LogLevel.CRITICAL);
+  });
+});
+
+describe('logLevelToString', () => {
+  it('should print log levels as string', () => {
+    chai.expect(logLevelToString(LogLevel.OFF)).to.equal('OFF');
+    chai.expect(logLevelToString(LogLevel.TRACE)).to.equal('TRACE');
+    chai.expect(logLevelToString(LogLevel.DEBUG)).to.equal('DEBUG');
+    chai.expect(logLevelToString(LogLevel.INFO)).to.equal('INFO');
+    chai.expect(logLevelToString(LogLevel.WARN)).to.equal('WARN');
+    chai.expect(logLevelToString(LogLevel.ERROR)).to.equal('ERROR');
+    chai.expect(logLevelToString(LogLevel.CRITICAL)).to.equal('CRITICAL');
+    chai.expect(logLevelToString(100.0)).to.been.undefined;
+  });
+});
+
+describe('LogEvent', () => {
+  it('should capture all constructor arguments', () => {
+    const level = LogLevel.WARN;
+    const category = 'Event-Category';
+    const message = 'Event-Message';
+    const properties = { authenticated: true };
+    const metrics = { duration: 23 };
+    const event = new LogEvent(level, category, message, properties, metrics);
+    event.level.should.be.equal(level);
+    event.category.should.be.equal(category);
+    event.message.should.be.equal(message);
+    event.properties.should.be.equal(properties);
+    event.metrics.should.be.equal(metrics);
+  });
+});
+
+describe('LoggingFacade', () => {
+  it('should not be able to be instantiated', () => {
+    (() => new LoggingFacade()).should.throw(Error);
+  });
+});
+
+describe('LoggingFacadeImpl', () => {
+  it('should be able to be instantiated', () => {
+    (new LoggingFacadeImpl(() => {})).should.been.instanceOf(LoggingFacadeImpl);
```

Comment:
been? is that part of chai?

Reply:
It is a mistake but sinon for chai introduces the chain been as same to be so it didn't failed.